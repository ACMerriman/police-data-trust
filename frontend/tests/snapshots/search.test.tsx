import * as React from "react"
import Dashboard from "../../pages/search"
import { render, router, setAuthForTest, userEvent, waitFor } from "../test-utils"

beforeAll(() => setAuthForTest())

it("renders Dashboard correctly", async () => {
  const { container } = render(<Dashboard />)
  expect(container).toMatchSnapshot()
})

it("redirects to login on logout", async () => {
  const { getByRole } = render(<Dashboard />)
  const logout = getByRole("button", { name: /logout/i })

  userEvent.click(logout)

  await waitFor(() => expect(router.pathname).toBe("/login"))
})
