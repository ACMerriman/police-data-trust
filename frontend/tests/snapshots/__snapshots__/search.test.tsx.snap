// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Dashboard correctly 1`] = `
<div>
  <div>
    <header
      class="wrapper"
    >
      <div
        class="backgroundBanner"
      >
        <div
          class="leftHeader"
        >
          <img
            alt="A blue circular logo design reads 'National Police Data Coalition'"
            class="logo medium"
            src="/NPDCLogo.svg"
          />
          <div
            class="titleContainer"
          >
            <h2>
              National Police Data Coalition
            </h2>
            <p>
              The national index of police incidents
            </p>
          </div>
        </div>
        <nav
          aria-label="Main Navigation"
        >
          <ul
            id="navMenu"
          >
            <li
              class=""
            >
              <a
                href="/search"
                role="menuitem"
                tabindex="-1"
              >
                Search
              </a>
            </li>
            <li
              class=""
            >
              <a
                href="/profile"
                role="menuitem"
                tabindex="-1"
              >
                Profile
              </a>
            </li>
            <li
              class=""
            >
              <a
                href="/logout"
                role="menuitem"
                tabindex="-1"
              >
                Sign Out
              </a>
            </li>
            <li
              class=""
            >
              <a
                href="/"
                role="menuitem"
                tabindex="-1"
              >
                DONATE
              </a>
            </li>
          </ul>
        </nav>
      </div>
    </header>
    <div
      style="display: flex; flex-direction: row; justify-content: center;"
    >
      <button
        class="primaryButton"
        style="cursor: pointer; margin: 2em 1em 1em 1em;"
        type="button"
      >
        bubble
      </button>
      <button
        class="primaryButton"
        style="cursor: pointer; margin: 2em 1em 1em 1em;"
        type="button"
      >
        map
      </button>
    </div>
    <div
      class="mapContainer"
      id="map-container"
    >
      <div
        class="mapWrapper"
      >
        <svg
          class="mapData"
          id="map-svg"
          viewBox="0, 0, 1200, 700"
        >
          <g
            id="zoom-container"
          >
            <svg
              id="map"
              viewBox="0, 0, 1200, 700"
            />
            <svg
              class="markerLayer"
              height="100%"
              viewBox="0, 0, 1200, 700"
              width="100%"
            />
          </g>
        </svg>
      </div>
      <div
        class="mapButton"
      >
        <button
          class="primaryButton"
          id="zoom-reset"
        >
          Reset Zoom
        </button>
      </div>
      <div
        style="position: absolute; right: 0px; bottom: 0px;"
      >
        <div
          class="chartKeyContainer"
          id="map-key"
        >
          <div
            class="KeyEntry"
          >
            <div
              class="keySymbolEntry"
            >
              <svg
                height="20"
                viewBox="0 0 20 20"
                width="20"
              >
                <circle
                  cx="10"
                  cy="10"
                  fill="var(--darkBlue)"
                  r="10"
                />
              </svg>
              <label
                class="label"
                for="Data Rich Cities"
              >
                Data Rich Cities
              </label>
            </div>
          </div>
          <div
            class="KeyEntry"
          >
            <div
              class="keySymbolEntry"
            >
              <svg
                height="20"
                viewBox="0 0 20 20"
                width="20"
              >
                <rect
                  fill="var(--lightBlue)"
                  height="20"
                  width="20"
                />
              </svg>
              <label
                class="label"
                for="Data Poor Cities"
              >
                Data Poor Cities
              </label>
            </div>
          </div>
          <div
            class="KeyEntry"
          >
            <div />
            <div
              class="keyGradientEntry"
            >
              <label
                class="label"
              >
                Less Data
              </label>
              <svg
                height="20"
                viewBox="0 0 20 20"
                width="20"
              >
                <rect
                  fill="rgb(124, 174, 215)"
                  height="20"
                  width="20"
                />
              </svg>
              <svg
                height="20"
                viewBox="0 0 20 20"
                width="20"
              >
                <rect
                  fill="rgb(105, 144, 186)"
                  height="20"
                  width="20"
                />
              </svg>
              <svg
                height="20"
                viewBox="0 0 20 20"
                width="20"
              >
                <rect
                  fill="rgb(86, 113, 157)"
                  height="20"
                  width="20"
                />
              </svg>
              <svg
                height="20"
                viewBox="0 0 20 20"
                width="20"
              >
                <rect
                  fill="rgb(67, 83, 128)"
                  height="20"
                  width="20"
                />
              </svg>
              <svg
                height="20"
                viewBox="0 0 20 20"
                width="20"
              >
                <rect
                  fill="rgb(48, 52, 99)"
                  height="20"
                  width="20"
                />
              </svg>
              <label
                class="label"
              >
                More Data
              </label>
            </div>
          </div>
          <div
            class="KeyEntry"
          >
            <div />
            <div
              class="keyGradientEntry"
            >
              <label
                class="label"
              >
                Blue
              </label>
              <svg
                height="20"
                viewBox="0 0 20 20"
                width="20"
              >
                <circle
                  cx="10"
                  cy="10"
                  fill="rgb(124, 174, 215)"
                  r="10"
                />
              </svg>
              <svg
                height="20"
                viewBox="0 0 20 20"
                width="20"
              >
                <circle
                  cx="10"
                  cy="10"
                  fill="rgb(119, 156, 187)"
                  r="10"
                />
              </svg>
              <svg
                height="20"
                viewBox="0 0 20 20"
                width="20"
              >
                <circle
                  cx="10"
                  cy="10"
                  fill="rgb(113, 138, 159)"
                  r="10"
                />
              </svg>
              <svg
                height="20"
                viewBox="0 0 20 20"
                width="20"
              >
                <circle
                  cx="10"
                  cy="10"
                  fill="rgb(108, 120, 130)"
                  r="10"
                />
              </svg>
              <svg
                height="20"
                viewBox="0 0 20 20"
                width="20"
              >
                <circle
                  cx="10"
                  cy="10"
                  fill="rgb(102, 102, 102)"
                  r="10"
                />
              </svg>
              <label
                class="label"
              >
                Grey
              </label>
            </div>
          </div>
        </div>
      </div>
    </div>
    <section
      style="max-width: 500px; margin: 1em auto;"
    >
      <h1>
        Search Incidents
      </h1>
      <form
        class="centerContent"
      >
        <fieldset
          style="display: flex; align-items: center;"
        >
          <label>
            Location
          </label>
          <input
            name="Location"
            placeholder="Location keyword"
            style="margin: 10px; flex-basis: auto;"
            type="text"
          />
        </fieldset>
        <fieldset
          style="display: flex; align-items: center;"
        >
          <label>
            Start Date
          </label>
          <input
            name="Start Date"
            style="margin: 10px; flex-basis: auto;"
            type="date"
          />
        </fieldset>
        <fieldset
          style="display: flex; align-items: center;"
        >
          <label>
            End Date
          </label>
          <input
            name="End Date"
            style="margin: 10px; flex-basis: auto;"
            type="date"
          />
        </fieldset>
        <fieldset
          style="display: flex; align-items: center;"
        >
          <label>
            Description
          </label>
          <input
            name="Description"
            placeholder="Description keyword"
            style="margin: 10px; flex-basis: auto;"
            type="text"
          />
        </fieldset>
        <button
          class="primaryButton"
          type="submit"
        >
          Search
        </button>
      </form>
    </section>
  </div>
</div>
`;
